be/src/exec/vectorized/hudi_mor_scanner.cpp

```diff
@@ -0,0 +1,348 @@
+            "instant_time -> {}, data_file_path -> {}, data_file_lenth -> {}, delta_log_paths count -> {}, serde_lib "
+            "-> {}, input_format -> {}",
+            _scan_ctx.base_path, _scan_ctx.hive_column_names, _scan_ctx.hive_column_types,
+            _scan_ctx.required_fields.size(), _scan_ctx.instant_time, _scan_ctx.data_file_path,
+            _scan_ctx.data_file_lenth, _scan_ctx.delta_log_paths.size(), _scan_ctx.serde_lib, _scan_ctx.input_format);
+    for (int i = 0; i < _scan_ctx.required_fields.size(); i++) {
+        message.append(fmt::format(" required_field{} -> {},", i, _scan_ctx.required_fields[i]));
+    }
+    for (int i = 0; i < _scan_ctx.delta_log_paths.size(); i++) {
+        message.append(fmt::format(" log{} -> {},", i, _scan_ctx.delta_log_paths[i]));
+    }
+    LOG(INFO) << message;
+
+    int fetch_size = state->chunk_size();
+    _hudi_mor_scanner = _jni_env->NewObject(
+            _hudi_mor_scanner_cls, constructor, jbase_path, jhive_column_names, jhive_column_types, jrequired_fields,
+            jinstant_time, jlog_paths, jdata_file_path, _scan_ctx.data_file_lenth, jserde, jinput_format, fetch_size);
+
+    DCHECK(_hudi_mor_scanner != nullptr);
+    CHECK_JAVA_EXCEPTION("get hudi MOR scanner failed")
+    // init jmethod
+    _scanner_has_next = _jni_env->GetMethodID(_hudi_mor_scanner_cls, "hasNext", "()Z");
+    DCHECK(_scanner_has_next != nullptr);
+    _scanner_get_next_chunk = _jni_env->GetMethodID(_hudi_mor_scanner_cls, "nextChunkOffHeap", "()J");
+    DCHECK(_scanner_get_next_chunk != nullptr);
+    _scanner_column_release = _jni_env->GetMethodID(_hudi_mor_scanner_cls, "releaseOffHeapColumnVector", "(I)V");
+    DCHECK(_scanner_column_release != nullptr);
+
+    _jni_env->DeleteLocalRef(jbase_path);
+    _jni_env->DeleteLocalRef(jhive_column_names);
+    _jni_env->DeleteLocalRef(jhive_column_types);
+    _jni_env->DeleteLocalRef(jrequired_fields);
+    _jni_env->DeleteLocalRef(jinstant_time);
+    _jni_env->DeleteLocalRef(jlog_paths);
+    _jni_env->DeleteLocalRef(jdata_file_path);
+    _jni_env->DeleteLocalRef(jserde);
+    _jni_env->DeleteLocalRef(jinstant_time);
+    _jni_env->DeleteLocalRef(jstring_class);
+    _jni_env->DeleteLocalRef(hudi_mor_scanner_util_cls);
+    return Status::OK();
+}
+
+Status HudiMORScanner::_has_next(bool* result) {
+    jboolean ret = _jni_env->CallBooleanMethod(_hudi_mor_scanner, _scanner_has_next);
+    CHECK_JAVA_EXCEPTION("call hudi MOR scanner hasNext failed")
+    *result = ret;
+    return Status::OK();
+}
+
+Status HudiMORScanner::_get_next_chunk(long* chunk_meta) {
+    SCOPED_TIMER(_profile.io_timer);
+    COUNTER_UPDATE(_profile.io_counter, 1);
+    *chunk_meta = _jni_env->CallLongMethod(_hudi_mor_scanner, _scanner_get_next_chunk);
+    CHECK_JAVA_EXCEPTION("getNextChunk failed")
+    return Status::OK();
+}
+
+template <PrimitiveType type, typename CppType>
+void HudiMORScanner::_append_data(Column* column, CppType& value) {
```

Comment:
Plz extract the entire converter logic to independent class

Reply:
done